### PR TITLE
Improve test coverage for struct search

### PR DESCRIPTION
**Pull request by ttnghia**
This PR adds a simple test case for struct binary search. In this test, we do binary search for 2 structs columns in which one column has a bit mask (but no null element) while the other column does not have any bit mask.

Reference: https://github.com/rapidsai/cudf/pull/8374 | https://github.com/rapidsai/cudf/issues/8187

### CODE REVIEW

File: cpp/tests/search/search_struct_test.cpp [hunk 1]
```diff
@@ -1,316 +1,355 @@
...
+  // Sorted dsc, search nulls last
+  results =
+    search_bounds(structs_t, structs_values, {cudf::order::DESCENDING}, {cudf::null_order::AFTER});
+  expected_lower_bound = int32s_col{7, 0, 0, 0, 3, 0};
+  expected_upper_bound = int32s_col{8, 0, 0, 0, 7, 0};
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected_lower_bound, results.first->view(), print_all);
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected_upper_bound, results.second->view(), print_all);
+}
+
+TYPED_TEST(TypedStructSearchTest, OneColumnHasNullMaskButNoNullElementTest)
```
Comment (ttnghia) [inline cpp/tests/search/search_struct_test.cpp:241]: I don't know why github can't show diff at this time. In this PR, only this function is added. Nothing else has been changed.

File: cpp/tests/search/search_struct_test.cpp [hunk 2]
```diff
@@ -1,316 +1,355 @@
...
+  structs_t   = structs_col{{child_col_t}, null_at(10)}.release();
+  results =
+    search_bounds(structs_t, structs_values, {cudf::order::DESCENDING}, {cudf::null_order::AFTER});
+  expected_lower_bound = int32s_col{7, 8, 0, 8, 3, 0};
+  expected_upper_bound = int32s_col{8, 11, 0, 11, 7, 0};
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected_lower_bound, results.first->view(), print_all);
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected_upper_bound, results.second->view(), print_all);
+}
+
+TYPED_TEST(TypedStructSearchTest, OneColumnHasNullMaskButNoNullElementTest)
```
Comment (ttnghia) [inline cpp/tests/search/search_struct_test.cpp:290]: I don't know why github can't show diff at this time. In this PR, only this function is added. Nothing else has been changed.

**Review by codereport — APPROVED** [review]
lgtm :+1:

**Comment by ttnghia** [comment]
Rerun tests.

**Comment by ttnghia** [comment]
Rerun tests.

**Comment by ttnghia** [comment]
Rerun tests.

**Comment by ttnghia** [comment]
@gpucibot merge

**Pull request merged**